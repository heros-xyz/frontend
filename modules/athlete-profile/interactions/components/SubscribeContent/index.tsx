import { Box, Flex, Link, Text } from "@chakra-ui/react";
import { FC } from "react";

interface ISubscribeContentProps {
  athleteName: string;
  onClick: () => void;
}

const SubscribeContent: FC<ISubscribeContentProps> = ({
  athleteName,
  onClick,
}) => {
  return (
    <Flex bg="accent.2" borderRadius={{ base: "md", lg: "12px" }}>
      <Box
        color="white"
        py={{ base: "3" }}
        px={{ base: "4" }}
        p={{ lg: "30px" }}
      >
        <Text fontWeight="bold" fontSize={{ base: "sm", lg: "xl" }}>
          Subcribe to exclusive content!
        </Text>
        <Text fontWeight="normal" fontSize={{ base: "xs", lg: "md" }}>
          You can view fan-only posts and interact with all{" "}
          <b>{athleteName}â€™s </b>
          content once you are a sucbriber.
        </Text>
      </Box>
      <Flex alignItems="center" borderLeft="1px white solid">
        <Link
          px="7"
          fontWeight="bold"
          fontSize={{ base: "sm", lg: "xl" }}
          color="white"
          onClick={onClick}
        >
          Subscribe
        </Link>
      </Flex>
    </Flex>
  );
};
export default SubscribeContent;
