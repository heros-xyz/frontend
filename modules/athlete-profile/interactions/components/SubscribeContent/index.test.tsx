import { render } from "@testing-library/react";
import SubscribeContent from ".";

it("renders subscribe content section unchanged", () => {
  const { container } = render(<SubscribeContent />);
  expect(container).toMatchSnapshot();
});
