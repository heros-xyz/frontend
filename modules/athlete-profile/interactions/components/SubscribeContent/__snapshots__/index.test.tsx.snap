// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders subscribe content section unchanged 1`] = `
<div>
  <div
    class="css-1vwpc76"
  >
    <div
      class="css-5bd7x2"
    >
      <p
        class="chakra-text css-1tdozh"
      >
        Subcribe to exclusive content!
      </p>
      <p
        class="chakra-text css-13oolba"
      >
        You can view fan-only posts and interact with all
         
        <b>
          â€™s 
        </b>
        content once you are a sucbriber.
      </p>
    </div>
    <div
      class="css-1onq7gh"
    >
      <a
        class="chakra-link css-1kjg9rc"
      >
        Subscribe
      </a>
    </div>
  </div>
</div>
`;
